haystack/nodes/other/join_docs.py

```diff
@@ -60,7 +72,17 @@ def run_accumulated(self, inputs: List[dict], top_k_join: Optional[int] = None):
         else:
             raise ValueError(f"Invalid join_mode: {self.join_mode}")
 
-        sorted_docs = sorted(scores_map.items(), key=lambda d: d[1], reverse=True)
+        # only sort the docs if that was requested
+        if self.sort_by_score:
+            sorted_docs = sorted(scores_map.items(), key=lambda d: d[1] if d[1] is not None else -inf, reverse=True)
+            if any(1 for s in scores_map.values() if s is None):
```

Comment:
Small simplification: `any` evaluates a list of booleans, so you don't need the `if`. It's also much closer to natural language.

```suggestion
            if any(s is None for s in scores_map.values()):
```

Reply:
Thanks @ZanSara , you are right. I have made the recommended simplification.
I hope we are good to merge now.
Thanks!